pkg/cluster/cluster.go

```diff
@@ -184,5 +187,6 @@ func New(ctx context.Context, log *logrus.Entry, _env env.Interface, db database
 		hiveClusterManager:                hiveClusterManager,
 		now:                               func() time.Time { return time.Now() },
 		openShiftClusterDocumentVersioner: new(openShiftClusterDocumentVersionerService),
+		byoNSG:                            feature.IsRegisteredForFeature(subscriptionDoc.Subscription.Properties, api.FeatureFlagBYONsg),
```

Comment:
if we set this in the cluster doc, we don't have to add the parameter to the manager struct.